lib/ecto/adapters/mysql.ex

```diff
@@ -196,6 +196,34 @@ defmodule Ecto.Adapters.MySQL do
   end
 
   @doc false
+  def upsert(repo, %{source: {prefix, source}, autogenerate_id: {key, :id}}, params, on_conflict, conflict_target, update, returning, opts) do
```

Comment:
This line is too long. You can call the second argument `meta` and do the matching on the key and source on the next line.